src/spv/bitcoin/BRWallet.h

```diff
@@ -56,11 +58,19 @@ inline static int BRUTXOEq(const void *utxo, const void *otherUtxo)
                                   ((const BRUTXO *)utxo)->n == ((const BRUTXO *)otherUtxo)->n));
 }
 
+
+// Comparator for C++ containers of UInt160
+bool UInt160Compare(const UInt160& a, const UInt160& b);
+
+// Convert SPV UInt256 to Bitcoin uint256
+uint256 to_uint256(UInt256 const & i);
+
 typedef struct BRWalletStruct BRWallet;
 
 // allocates and populates a BRWallet struct that must be freed by calling BRWalletFree()
 // forkId is 0 for bitcoin, 0x40 for b-cash
-BRWallet *BRWalletNew(BRTransaction *transactions[], size_t txCount, BRMasterPubKey mpk, int forkId);
+BRWallet *BRWalletNew(BRTransaction *transactions[], size_t txCount, BRMasterPubKey mpk, int forkId,
+                      std::set<UInt160, decltype(&UInt160Compare)>&& userAddresses);
```

Comment:
Can define a type for `std::set<UInt160, decltype(&UInt160Compare)>`? I see it used in quite a lot places.

Reply:
It's a Lambda comparator.

Reply:
I'll update with a struct.

Reply:
@ShengguangXiao https://github.com/DeFiCh/ain/pull/282/commits/eaf75eff44be33b9139bf99a2a2f25dfc6108746

Reply:
Actually what i mean is make std::set<UInt160, UInt160Compare> as SetOfAddress like below, but i think now is quite simple already. Thanks.
```
using SetOfAddress = std::set<UInt160, UInt160Compare>
```

Reply:
Oh right, I thought you did not like the somewhat cumbersome decltype(&UInt160Compare) as the comparator, I guess a using would have helped there. I often personally prefer to see what it is and not a using declaration in its place, with the decltype included a using declaration would have been better but I think it is okay now.